extensions/roc-package-web-app-react/src/app/server/reactRenderer.js

```diff
@@ -184,14 +210,21 @@ export function reactRender({
                     const reduxState = store ? store.getState() : {};
                     const status = ServerStatus.rewind() || 200;
 
-                    let customTemplateValues;
+                    let customTemplateValues = invokeHook('get-template-values', {
+                        koaState,
+                        settings: rocConfig,
+                        reduxState,
+                    });
                     if (hasTemplateValues) {
                         // Provides settings, Redux state and Koa state
-                        customTemplateValues = templateValues.default({
-                            koaState,
-                            settings: rocConfig,
-                            reduxState,
-                        });
+                        customTemplateValues = merge(
```

Comment:
It would probably be enough. Depends on the kind of merge behavior we want. Otoh, since we leave the "merge with previous values" up to the actions, perhaps a `Object.assign` is quite enough here?